addons/binding/org.openhab.binding.rotelra1x/src/main/java/org/openhab/binding/rotelra1x/handler/RotelRa1xHandler.java

```diff
@@ -0,0 +1,315 @@
+        }
+    }
+
+    private String readCommand() throws IOException {
+        int b;
+        StringBuilder commandPart = new StringBuilder();
+        do {
+            b = serialPort.getInputStream().read();
+            if (b == -1) {
+                throw new IOException("Connection unexpectedly closed");
+            } else {
+                commandPart.append((char) b);
+            }
+        } while (b != '!' && b != '=');
+        return commandPart.toString().substring(0, commandPart.length() - 1);
+    }
+
+    private String readUntil(char terminator) throws IOException {
+        int b;
+        StringBuilder commandPart = new StringBuilder();
+        do {
+            b = serialPort.getInputStream().read();
+            if (b == -1) {
+                throw new IOException("Connection unexpectedly closed");
+            } else {
+                commandPart.append((char) b);
+            }
+        } while (b != terminator);
+        return commandPart.toString().substring(0, commandPart.length() - 1);
+    }
+
+    public PercentType readVolume() throws IOException {
+        String volumeString = readUntil('!');
+        int volume;
+        if (volumeString.equals("min")) {
+            volume = 0;
+        } else if (volumeString.equals("max")) {
+            volume = max_vol;
+        } else {
+            volume = Integer.parseInt(volumeString);
+        }
+        double volumePct = volume * 100.0 / max_vol;
+        return new PercentType(BigDecimal.valueOf(Math.round(volumePct)));
+    }
+
+    public PercentType readDimmer() throws IOException {
+        String dimmerString = readUntil('!');
+        int dimmer = Integer.parseInt(dimmerString);
+        // Let's make 100 % the brightest, makes more sense
+        double dimmerPct = 100.0 - (dimmer * 100.0 / 6.0);
+        return new PercentType(BigDecimal.valueOf(Math.round(dimmerPct)));
+    }
+
+    public DecimalType readFrequency() throws IOException {
+        String freqString = readUntil('!');
+        double freq;
+        if ("off".equals(freqString)) {
+            freq = 0.0;
+        } else {
+            freq = Double.parseDouble(freqString);
+        }
+        return new DecimalType(freq);
+    }
+
+    void powerOnRefresh() {
+        new Thread() {
+            @Override
+            public void run() {
+                try {
+                    send("get_volume!");
+                    send("get_current_source!");
+                } catch (IOException e) {
+                    e.printStackTrace();
+                }
+            }
+        }.start();
+    }
+
+    @Override
+    public void run() {
+        try {
+            Thread.sleep(1000); // Seems we need to wait a bit after initialization or channels won't
+                                // be updated. (making run() and initialize() synchronized doesn't work)
+        } catch (InterruptedException e1) {
```

Comment:
If we receive this InterruptedException, the interrupted flag was cleared and we most likely need to reset the flag:

http://www.yegor256.com/2015/10/20/interrupted-exception.html
https://www.ibm.com/developerworks/library/j-jtp05236/